pkg/manifests/config.go

```diff
@@ -268,6 +268,7 @@ type OpenShiftStateMetricsConfig struct {
 type K8sPrometheusAdapter struct {
 	NodeSelector map[string]string `json:"nodeSelector"`
 	Tolerations  []v1.Toleration   `json:"tolerations"`
+	AuditLevel   string            `json:"auditLevel"`
```

Comment:
@dgrisonnet https://docs.openshift.com/container-platform/4.8/security/audit-log-policy-config.html#configuring-audit-policy_audit-log-policy-config  uses `profiles`. 

 We could do the same where each profile has a different meaning `Default, WriteRequestBodies, or AllRequestBodies`  but in our case, the default is to not enable any audit logs. I  prefer this implementation where the control is the `Level` of `Audit Logs` to theirs where each profile has a different meaning.  WDYT?

Reply:
Apparently there's also a `None` profile:

https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_apiserver.go#L68-L82

The advantage of using the same approach is that it reduces the cognitive load for users (e.g. consistent with the other OpenShift components) and we can easily link to the existing OpenShift docs to explain what each profile does. The flip side is that it may require more work for CMO though we can probably reuse [existing code](https://github.com/openshift/library-go/blob/fd8365501a47f82b85f5433f821ab047ba19011a/pkg/operator/apiserver/audit/audit_policies.go#L84-L114) to generate the policy configmap. Another disadvantage is that it would be harder/impossible to support the same in kube-prometheus.

Reply:
I also think that reusing the same format as the other OpenShift components would make it easier for the end-users, so I would side with using the same levels as the apiserver.

Instead of using a string to store the audit level we could use the same format as the apiserver and have an [Audit](https://github.com/openshift/library-go/blob/fd8365501a47f82b85f5433f821ab047ba19011a/vendor/github.com/openshift/api/config/v1/types_apiserver.go#L84-L116) object from which we could generate a profile based on that with [GetAuditPolicy](https://github.com/openshift/library-go/blob/fd8365501a47f82b85f5433f821ab047ba19011a/pkg/operator/apiserver/audit/audit_policies.go#L88-L114).

> Another disadvantage is that it would be harder/impossible to support the same in kube-prometheus.

Since these profiles are OpenShift specific, we won't be able to have the same in both upstream and downstream if we were to follow the route of consistency. That said, with the current implementation of kube-prometheus, I would not expect audit logs to be configured by default and only configurable in an addon. So I think it is fine even if we were to diverge here.

>  the default is to not enable any audit logs

Thinking about it again, I don't see any disadvantages to having audit logs by default. I think we should be consistent with the apiserver here and set it to `Default` profile by default. This would improve the observability of the apiserver as a whole and if audit logs were deemed valuable enough to be enabled by default in the apiserver, that should also be the case for aggregated APIs to cover all API resources.

Reply:
If we manage to find a solution for limiting the size of the logs, it's fine by me.

Reply:
We can set a sane default for the `--audit-log-maxsize` flag of prometheus-adapter. If the size is exceeded, the log file will be rotated.